### ISSUE TITLE
Confusing diagnostic for assignment with trailing comma

### ISSUE THREAD

**Issue by Gankra** [astral-sh/ty#1797]
https://play.ty.dev/2fdc0a03-ef37-44f1-8074-f8558ff7a582

```py
coords: tuple[int, int] = (0, 0),
```

> Object of type `tuple[tuple[Literal[0], Literal[0]]]` is not assignable to `tuple[int, int]` (invalid-assignment) 

I ran into this while turning a class into a `@dataclass`, and almost filed a bug about dataclasses not computing assignability correctly... and only while writing this just now did I process that that says `tuple[tuple[...`.

This message is totally correct (this value prints `((0, 0),)`) but I wonder if we could notice this situation and be like "hey you have a trailing comma, did you... mean that?"

**Comment by MichaReiser** [comment]
You should give ruff format a try
